core/src/main/java/io/aiven/klaw/service/OperationalRequestsService.java

```diff
@@ -0,0 +1,337 @@
+    return switch (order) {
+      case ASC_REQUESTED_TIME -> compareByTime();
+      case DESC_REQUESTED_TIME -> Collections.reverseOrder(compareByTime());
+    };
+  }
+
+  private static Comparator<OperationalRequest> compareByTime() {
+    return Comparator.comparing(OperationalRequest::getRequesttime);
+  }
+
+  public ApiResponse resetConsumerOffsets(String req_no) {
+    log.info("approveConsumerOffsetRequests {}", req_no);
+    final String userDetails = getUserName();
+    int tenantId = commonUtilsService.getTenantId(userDetails);
+    if (commonUtilsService.isNotAuthorizedUser(
+        getPrincipal(), PermissionType.APPROVE_SUBSCRIPTIONS)) {
+      return ApiResponse.NOT_AUTHORIZED;
+    }
+    ResetConsumerGroupOffsetsRequest resetConsumerGroupOffsetsRequest =
+        ResetConsumerGroupOffsetsRequest.builder().build();
+
+    HandleDbRequests dbHandle = manageDatabase.getHandleDbRequests();
+    OperationalRequest operationalRequest =
+        dbHandle.getOperationalRequest(Integer.parseInt(req_no), tenantId);
+    ApiResponse apiResponse;
+    try {
+      apiResponse =
+          clusterApiService.resetConsumerOffsets(
+              resetConsumerGroupOffsetsRequest, operationalRequest.getEnvironment(), tenantId);
+    } catch (KlawException e) {
+      return ApiResponse.notOk(ApiResultStatus.FAILURE.value);
+    }
+
+    if (apiResponse.isSuccess()) {
+      if (apiResponse.getData() instanceof Map) {
+        Map<OffsetsTiming, Map<String, Long>> offsetPositionsBeforeAndAfter =
+            (Map) apiResponse.getData();
+        String beforeReset =
+            "\n\nBefore Offset Reset"
+                + offsetPositionsBeforeAndAfter.get(OffsetsTiming.BEFORE_OFFSET_RESET);
+        String afterReset =
+            "\n\nAfter Offset Reset"
+                + offsetPositionsBeforeAndAfter.get(OffsetsTiming.AFTER_OFFSET_RESET);
+        String offsetResetDetails =
+            resetConsumerGroupOffsetsRequest.getConsumerGroup() + "\n" + beforeReset + afterReset;
+        mailService.sendMail(
+            operationalRequest.getTopicname(),
+            offsetResetDetails,
+            "",
+            operationalRequest.getRequestor(),
+            operationalRequest.getApprover(),
+            operationalRequest.getRequestingTeamId(),
+            dbHandle,
+            MailType.RESET_CONSUMER_OFFSET_APPROVED,
+            commonUtilsService.getLoginUrl());
+      }
+    }
+
+    return apiResponse.isSuccess() ? ApiResponse.SUCCESS : ApiResponse.FAILURE;
+  }
+
+  private void checkIsAuthorized(PermissionType permission) throws KlawNotAuthorizedException {
+    if (commonUtilsService.isNotAuthorizedUser(getPrincipal(), permission)) {
+      throw new KlawNotAuthorizedException(TOPICS_ERR_101);
+    }
+  }
+
+  private String getUserName() {
+    return mailService.getCurrentUserName();
+  }
+
+  private Object getPrincipal() {
+    return SecurityContextHolder.getContext().getAuthentication().getPrincipal();
+  }
+}
```

Comment:
Should these two be in a common class?

Reply:
I like to move them, also which exist in other classes. in a different PR though.